src/java/org/apache/cassandra/exceptions/InvalidColumnTypeException.java

```diff
@@ -85,20 +86,20 @@ public AbstractType<?> tryFix()
         return null;
     }
 
-    private AbstractType<?> tryFixInternal()
+    private AbstractType<?> tryFixInternal(boolean mayBeImplicitlyFrozen)
```

Comment:
A false `mayBeImplicitlyFrozen` makes this no-op. So we could just remove the argument from this class, and simply change the call in `SerializationHeader.Component#validateType` to:
```java
AbstractType<?> fixed = sstableVersion.hasExplicitlyFrozenTuples() ? null : e.tryFix();
```
And we would just skip the call in `SchemaKeyspace#validate`. In fact, that method can be removed and replaced by direct calls to `AbstractType#validateForColumn`.

Reply:
since that - I'd move `tryFix` out from the exception class to `SerializationHeader` and make it private